Configuration/Test/test.py

```diff
@@ -0,0 +1,534 @@
+
+process.load("SimG4Core.Application.g4SimHits_cfi")
+process.g4SimHits.Physics.BeamProtTransportSetup = BeamProtTransportSetup
+#process.g4SimHits.Generator.HepMCProductLabel = 'generator'    # The input source for G4 module is connected to "process.source".
+process.g4SimHits.G4TrackingManagerVerbosity = cms.untracked.int32(0)
+process.g4SimHits.OverrideUserStackingAction = cms.bool(True)   # HINT: TOTEM specific
+process.g4SimHits.TransportParticlesThroughWholeBeampipe = cms.bool(True)
+# TOTEM uses MeasuredGeometryRecord instead of IdealGeometryRecord
+process.g4SimHits.UseMeasuredGeometryRecord = cms.untracked.bool(False)  # HINT: TOTEM specific
+process.g4SimHits.SteppingVerbosity = cms.int32(0)
+process.g4SimHits.G4EventManagerVerbosity = cms.untracked.int32(0)
+process.g4SimHits.G4StackManagerVerbosity = cms.untracked.int32(0)
+process.g4SimHits.Watchers = cms.VPSet(
+#        cms.PSet( # HINT: TOTEM specific
+#            type = cms.string('SimTracer'),
+#            SimTracer = cms.PSet(verbose = cms.bool(True)),
+#        ),
+#       cms.PSet( # HINT: TOTEM specific
+#           type = cms.string('TotemRP'),
+#           TotemRP = cms.PSet(
+#               Names = cms.vstring('TotemHitsRP'),
+#               FileName = cms.string('TotemTestRP_Hits.root'),
+#               RPDebugFileName = cms.string('TotemDebugRP.root'),
+#               FileNameOLD = cms.string('TotemTestRP_Hits_Old.root'),
+#               Verbosity = cms.bool(True)
+#           )
+#       )
+    )
+process.g4SimHits.HepMCProductLabel = cms.InputTag("generator")
+process.g4SimHits.Physics.DefaultCutValue = cms.double(100.0)
+process.g4SimHits.Generator = cms.PSet(
+    HectorEtaCut,
+    HepMCProductLabel=cms.string('SmearingGenerator'),
+    ApplyPCuts=cms.bool(False),
+    ApplyPtransCut=cms.bool(False),
+    MinPCut=cms.double(0.04),  ## the cut is in GeV
+    MaxPCut=cms.double(99999.0),  ## the pmax=99.TeV
+    ApplyEtaCuts=cms.bool(False),
+    MinEtaCut=cms.double(-5.5),
+    MaxEtaCut=cms.double(5.5),
+    RDecLenCut=cms.double(2.9),  ## (cm) the cut on vertex radius
+    LDecLenCut=cms.double(30.0),  ## (cm) decay volume length
+    ApplyPhiCuts=cms.bool(False),
+    MinPhiCut=cms.double(-3.14159265359),  ## (radians)
+    MaxPhiCut=cms.double(3.14159265359),  ## according to CMS conventions
+    ApplyLumiMonitorCuts=cms.bool(False),  ## primary for lumi monitors
+    Verbosity=cms.untracked.int32(0),
+    LeaveScatteredProtons=cms.untracked.bool(True),
+    ## HINT: TOTEM specific - Leave intact protons after scattering for further near beam transport
+    LeaveOnlyScatteredProtons=cms.untracked.bool(False)
+    ## HINT: TOTEM specific - Leave only intact protons and reject all the other particles
+)
+
+process.g4SimHits.StackingAction.SaveFirstLevelSecondary = cms.untracked.bool(True)
+process.g4SimHits.Totem_RP_SD = cms.PSet(  # HINT: TOTEM specific
+    Verbosity=cms.int32(0)
+)
+process.g4SimHits.CastorSD.nonCompensationFactor = cms.double(0.85)
```

Comment:
Do we need that ? What is the default value of `process.g4SimHits.CastorSD.nonCompensationFactor` ?
Check "SimG4Core.Application.g4SimHits_cfi"

Reply:
Default value is 0.817 (SimG4Core.Application.g4SimHits_cfi)
It is inherited from Marcin repo: https://github.com/ziaber/cmssw/blob/simulation/Configuration/Test/test.py#L302
After applying some cout in CastorSD.cc and recompilation there is nothing in the output so I assume it is needless.

After commenting this out, and cmsRun for 100 events output file is 66B smaller.

Reply:
Remove this line, we do not simulate Castor detector

Reply:
done